src/Core/Block.cpp

```diff
@@ -494,6 +495,13 @@ DataTypes Block::getDataTypes() const
     return res;
 }
 
+static String getNameOfBaseColumn(const IColumn & column)
```

Comment:
Does this mean that if we have a Block with Sparse column, and another with fully materialized column, we will treat them as equal? Not sure this is correct, e.g. we don't do this for LowCardinality or Const, which are also different in-memory representations.

Reply:
I mean, the caller can probably get confused if it suddenly gets a Sparse column when it expects a normal one.

Reply:
Yes, blocks with sparse and full columns of the same type are treated as equal. Now kind of one column may differ between stream and in most cases it's ok (simple transforms, that uses only `IColumn` interface without `assert_cast` and transforms, where sparse columns are handled explicitly). I understand, that it became more dangerous now, and we need more accurate check.

Probably we can add two checks, one, that treats materialized/non-materialized columns equal, and other that doesn't. In places, where more strict check is needed (e.g. in `MergingSortedTransform`) , call the second version.